custom_report_label/models/custom_report_label.py

```diff
@@ -0,0 +1,38 @@
+# Copyright 2024 Quartile Limited
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class CustomReportLabel(models.Model):
+    _name = "custom.report.label"
+
+    name = fields.Char(required=True)
+    model_id = fields.Selection(
+        selection="_list_all_models", string="Model", required=True
+    )
+    template_id = fields.Many2one("ir.actions.report")
+    existing_label = fields.Char(
+        translate=True,
+        help="Enter the label to replace, including '>' and '<'.\n"
+        "Example: '>Description<'.",
+    )
+    custom_label = fields.Char(
+        translate=True,
+        help="Input your new label, including symbols.\n"
+        "Example: '>New Description<'. This replaces the existing label.",
+    )
+
+    @api.model
+    def _list_all_models(self):
+        lang = self.env.lang or "en_US"
+        self._cr.execute(
+            "SELECT model, COALESCE(name->>%s, name->>'en_US') FROM ir_model ORDER BY 2",
+            [lang],
+        )
+        vals = self._cr.fetchall()
+        return vals
```

Comment:
```suggestion
```
We prefer not to do this, but in case we want to create a list for selection, we should call the method in IrFilters instead of rewriting the same thing here.